lua/caramba/multifile.lua

```
@@ -388,55 +388,104 @@ end
+    content = table.concat(lines, "\n")
+
+    if not content:find(old_name, 1, true) then goto continue end
+
+    local lang = vim.filetype.match({filename = filepath})
+    if not lang or not require('nvim-treesitter.parsers').has_parser(lang) then goto continue end
+
+    local parser = vim.treesitter.get_string_parser(content, lang)
+    local tree = parser:parse()[1]
+    if not tree then goto continue end
+
+    local root = tree:root()
```

Luacheck (1.2.0)

[warning] shadowing definition of variable 'root' on line 394

(W421)